File: CPE/Exam/231212/G4_UVA993.py
T = int(input())

for _ in range(T):
	N = int(input())
	if N <= 1: # 0 or 1
		print(N)
		continue
	# 因數分解，檢查有沒有大於10的因數，注意可以合併2^3=8來縮小答案
	cnt = [0] * 10
	for p in range(9, 1, -1): # [9, 2]
		while (N % p == 0):
			cnt[p] += 1
			N //= p
	if N != 1: # 有大於10的因數
		print(-1)
	else:
		ans = ""
		for p in range(2, 10):
			ans += str(p) * cnt[p]
		print(ans)

File: CPE/Exam/231212/G4_UVA993.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
	ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
	int t, n, cnt[10];
	cin >> t;
	while (t--) {
		cin >> n;
		if (n <= 1) {
			cout << n << endl;
			continue;
		}
		memset(cnt, 0, sizeof(cnt)); 
		for (int p = 9; p > 1; p--) { // 因數分解
			while (n % p == 0) {
				cnt[p]++;
				n /= p;
			}
		}
		if (n != 1) { // 有大於10的因數
			cout << -1 << endl;
		} else {
			string ans = "";
			for (int p = 2; p < 10; p++) {
				while (cnt[p] --) {
					cout << p;
				}
			}
			cout << endl;
		}
	}
	return 0;
}